runtime/ts/storage/firebase-storage.ts

```diff
@@ -354,8 +354,8 @@ class FirebaseVariable extends FirebaseStorageProvider {
   private value: {storageKey: string, id: string}|null;
   private localModified: boolean;
   private readonly initialized: Promise<void>;
-  // TODO(sjmiles): localId collisions occur when using device-client-pipe,
-  // so I'll randomize localId a bit
+  // TODO(sjmiles): localKeyId collisions occur when using device-client-pipe,
+  // so I'll randomize localKeyId a bit
   private localKeyId = Date.now();
```

Comment:
Math.random() by itself scares me because it's deterministic, but I think I know less about this stuff that most of the rest of the team. :)

Also fwiw, I recent checked in some id-generation code using Firebase techniques: https://github.com/sjmiles/arcs/blob/master/shell_0_6_0/components/generate-id.js

Reply:
Thanks mike. Seems reasonable, just thought I'd ask in case there was a danger of growing an over reliance on this implementation.